app/services/onlinetesting/Phase3TestService.scala

```diff
@@ -185,70 +186,80 @@ trait Phase3TestService extends OnlineTestService with Phase3TestConcern {
       emailInviteToApplicant(application, emailAddress, invitationDate, expirationDate)
     }
 
-    def extendIfInvigilated(application: OnlineTestApplication): Future[Unit] = if (application.isInvigilatedVideo) {
-      extendTestGroupExpiryTime(
-        application.applicationId,
-        gatewayConfig.phase3Tests.invigilatedTimeToExpireInDays - daysUntilExpiry,
-        "InvigilatedInviteSystem"
-      )
-    } else {
-      Future.successful(())
+    def extendIfInvigilatedOrIfAdjustmentsWereChanged(application: OnlineTestApplication): Future[Unit] = {
+      def couldHaveBeenInvigilatedBefore(phase3TestGroup: Option[Phase3TestGroup]): Boolean =
+        phase3TestGroup.map(_.expirationDate.isAfter(dateTimeFactory.nowLocalTimeZone.plusDays(daysUntilExpiry*2))).getOrElse(false)
+
+      if (couldHaveBeenInvigilatedBefore(phase3TestGroup)) {
```

Comment:
I think something like this ends up just as readable, and saves the repetition of future.success

      if (couldHaveBeenInvigilatedBefore && !application.isInvigilatedVideo) {
          extendTestGroupExpiryTime(
            application.applicationId,
            - gatewayConfig.phase3Tests.invigilatedTimeToExpireInDays + daysUntilExpiry,
            "NonInvigilatedInviteSystem"
          )
      } else if (!couldHaveBeenInvigilatedBefore && application.isInvigilatedVideo) {
          extendTestGroupExpiryTime(
            application.applicationId,
            gatewayConfig.phase3Tests.invigilatedTimeToExpireInDays - daysUntilExpiry,
            "InvigilatedInviteSystem"
          )
      } else {
        Future.successful(())
      }